Update rule.md (#5468)
SSWConsulting/SSW.Rules.Content: rules/make-sure-that-the-test-can-be-failed/rule.md

---

> It's important that the unit tests you develop are capable of failing and that you have seen it fail. A test that can never fail isn't helpful for anyone.

Reviewer: ```suggestion
It's important that the unit tests you develop are capable of failing now or in future.  A test that can never EVER fail isn't helpful for anyone.
```

A `NotImplementedException` is synonymous with "still in development", include a `//TODO: ` marker with some notes about the steps to take to implement the test.

A better approach would be to return a value that is invalid.

```cs
[Test]
public void ShouldCheckIfPositive()
{
   var calculator = new Calculator();
   var result = calculator.IsPositive(10);

   Assert.True(result);
}

// The method to test in class Calculator ...
public int IsPositive(int x)
{
   return -1;
}
```

::: Good
Good Example: The test fails by returning an invalid value.
:::

::: Note
Sometimes there is no clear definition of an invalid value, then it is acceptable to fail a test using `NotImplementedException`. Add additional remarks, notes or steps on what to test and how to implement with a `//TODO: ...` marker. This will asssist you or other developers coming across this failed test. 

Make sure that this test will be implemented before a production release.
:::

```cs
// The method to test in class Calculator ...
public int IsPositive(int x)
{
   //NOTE: ths method has a clear "invalid" value
   return -1;
}
public int Sum(int x, int y)
{

>    //NOTE: this method does not have a clear "invalid" value and throws a NotImplementedException and inclused a TODO marker

Reviewer 2: ```suggestion
   //NOTE: this method does not have a clear "invalid" value and throws a NotImplementedException and includes a TODO marker
```
